update-format-crau/src/bin/test.rs

```diff
@@ -68,23 +78,104 @@ fn main() -> Result<(), Box<dyn Error>> {
     // this may also be the case for the InstallOperations
     // use header.translate_offset()
 
+    let mut siglen: u64 = 0;
+
     let signatures_bytes = match (manifest.signatures_offset, manifest.signatures_size) {
         (Some(sig_offset), Some(sig_size)) => {
             f.seek(SeekFrom::Start(header.translate_offset(sig_offset)))?;
 
             let mut buf = vec![0u8; sig_size as usize];
+            siglen = sig_size;
             f.read_exact(&mut buf)?;
             Some(buf.into_boxed_slice())
         }
         _ => None,
     };
 
-    let signatures = match signatures_bytes {
-        Some(ref bytes) => Some(proto::Signatures::parse_from_bytes(bytes)?),
-        None => None,
+    Ok(SigBuffer {
+        data: Box::leak(signatures_bytes.unwrap()),
```

Comment:
why leak the box here instead of just returning `Box<[u8]>` or `Vec<u8>`?

Reply:
Right. Will simply return `Box<[u8]>`.